lib/cdo/web_purify.rb

```diff
@@ -23,24 +51,33 @@ def self.find_potential_profanities(text, language_codes = ['en'])
       map {|code| ISO_639_1_TO_WEBPURIFY[code] || code}.
       uniq.
       join(',')
-    url = "http://api1.webpurify.com/services/rest/" \
-      "?api_key=#{CDO.webpurify_key}" \
-      "&method=webpurify.live.return" \
-      "&text=#{CGI.escape(text)}" \
-      "&lang=#{language_codes}" \
-      "&format=json"
-    result = JSON.
-      parse(
-        URI.parse(url).
-        open(
-          open_timeout: DCDO.get('webpurify_tcp_connect_timeout', 5),
-          read_timeout: DCDO.get('webpurify_http_read_timeout', 10)
-        ).
-        read
-      )
-
-    expletive = result['rsp'] && result['rsp']['expletive']
-    return nil unless expletive
-    expletive.is_a?(Array) ? expletive : [expletive]
+
+    chunks = split_text(text)
+    expletives = []
+    Net::HTTP.start(API_ENDPOINT.host, API_ENDPOINT.port, CONNECTION_OPTIONS) do |http|
+      chunks.each do |chunk|
+        request = Net::HTTP::Post.new(API_ENDPOINT)
+        form_data = [
+          ['api_key', CDO.webpurify_key],
+          ['format', 'json'],
+          ['lang', language_codes],
+          ['method', 'webpurify.live.return'],
+          ['text', chunk]
+        ]
+        request.set_form form_data, 'multipart/form-data'
+
+        response = http.request(request)
+
+        next unless response.is_a?(Net::HTTPSuccess)
```

Comment:
How do we want to handle errors here? It doesn't seem like there was explicit error handling before. (There was some client-side logic to render a useful message specifically for the large payload error.)

Reply:
The question here is whether we want to allow publishing if webpurify is down or something, yeah? I think that's maybe worth clarifying with the group while we're here.

Reply:
In general our error handling leaves something to be desired. I think it's worth reporting back a useful retry message if there is an issue communicating with WebPurify.